domain/infosync/info.go

```diff
@@ -330,6 +330,10 @@ func doRequest(ctx context.Context, addrs []string, route, method string, body i
 		}
 
 		res, err = util2.InternalHTTPClient().Do(req)
+
+		if strings.Contains(route, "placement-rule") && res.StatusCode == http.StatusNotFound {
```

Comment:
the `strings.Contains(route, "placement-rule")` condition seems unnecessary since `doRequest` is only used for placement-rule requests 🤔 

(BTW I think line 345 should be changed to check for the code HTTP 412 `http.StatusPreconditionFailed` instead of the exact string `"placement rules feature is disabled"`)

Reply:
> the strings.Contains(route, "placement-rule") condition seems unnecessary since doRequest is only used for placement-rule requests 🤔
For now, it's only used for placement-rule, but later it may be used by other requests?

> and this also affects GetRuleBundle() and PutRuleBundles(), I assume it is also OK to ignore errors for them due to line 345?
Hmmm, I don't quite understand it.

> (BTW I think line 345 should be changed to check for the code HTTP 412 http.StatusPreconditionFailed instead of the exact string "placement rules feature is disabled")
agree

Reply:
> For now, it's only used for placement-rule, but later it may be used by other requests?

The other requests in the future will be targeting new APIs for sure, so the same 404 issue will arise, and I bet the treatment is the same here - produce a no-op response.

(And other existing HTTP calls don't use `doRequest`, they seem to use `http.NewRequest` directly 🤔)

> Hmmm, I don't quite understand it.

Line 345 specializes the "placement rules feature is disabled" error and turn it into no-op. It affects all of GetRuleBundle and PutRuleBundles and GetAllRuleBundles. I mean this PR is making the HTTP 404 behavior the same as HTTP 412.